### PR TITLE
modules: cmsis: move _glue_ code to modules/cmsis

### PR DESCRIPTION
**Pull request by gmarull**
Move the CMSIS _glue_ code to modules/cmsis. Refer to commits for more details.

### CODE REVIEW

File: modules/cmsis/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# Copyright (c) 2023 Nordic Semiconductor ASA
+# SPDX-License-Identifier: Apache-2.0
+
+add_subdirectory(${ZEPHYR_CURRENT_MODULE_DIR} cmsis)
+zephyr_include_directories(.)
```
Comment (aescolar) [inline modules/cmsis/CMakeLists.txt:5]: Adding it like this unconditionally, it gets added to all builds.
So, for simulation, now we depend on the include path order. How things stand right now, this path is after the nrf52_bsim cmsis/ include, which means things did not blow up.
I'd add a ifdef ARM archs here or so, as this headers are the CMSIS Core glue.

**Review by aescolar — DISMISSED** [review]

File: modules/cmsis/CMakeLists.txt (hunk 1 above)
Comment (gmarull) [reply modules/cmsis/CMakeLists.txt:5]: the include path is unconditional, but same was before with the include found in `zephyr/arch/...`, right? I'm not sure I understand the problem here.

Comment (aescolar) [reply modules/cmsis/CMakeLists.txt:5]: That before it was added only if you were building with the ARM archs. Now it is always.

Comment (gmarull) [reply modules/cmsis/CMakeLists.txt:5]: As is today, cmsis _glue_ code resides in:

```
include/zephyr/arch/arm/aarch32/cortex_a_r/cmsis.h
include/zephyr/arch/arm/aarch32/cortex_m/cmsis.h
```

Which is always in the include path, no?

Comment (aescolar) [reply modules/cmsis/CMakeLists.txt:5]: Only for ARM targets

Comment (gmarull) [reply modules/cmsis/CMakeLists.txt]: ok, got the problem, bsim has its own `cmsis.h` in the path as well. should now be fixed with `cmsis_core.h` (also, include path is only added if actually makes sense, ie for M/A/R)

**Comment by gmarull** [comment]
@SgrrZhf rebased

**Review by ifyall — DISMISSED** [review]

**Comment by MaureenHelm** [comment]
Does this break out of tree SoCs?

**Comment by gmarull** [comment]
> Does this break out of tree SoCs?

Yes, header can be kept with a warning though if preferred.

**Comment by gmarull** [comment]
@MaureenHelm added https://github.com/zephyrproject-rtos/zephyr/pull/59855/commits/eab032750b2cc6c59d7db02d4f1516722c3f2ca5, it's a trivial case that will help.

**Review by carlocaione — DISMISSED** [review]

**Review by SgrrZhf — DISMISSED** [review]

**Review by microbuilder — DISMISSED** [review]

**Review by carlescufi — DISMISSED** [review]

**Comment by gmarull** [comment]
last push: updated module SHAs

**Review by microbuilder — APPROVED** [review]

**Review by fabiobaltieri — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**